### PR TITLE
add DI lib

### PR DESCRIPTION
**Pull request by sjwsl**
add lib for calculate DI

### CODE REVIEW

File: di/di.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+func getLabels(db *sql.DB, issue Issue) (map[string][]string, error) {
+	if db == nil {
+		return nil, errors.New("db is nil")
+	}
+
+	labels := make(map[string][]string)
+
+	ctx, cancel := context.WithTimeout(context.Background(), MYSQL_QUERY_TIMEOUT)
+	defer cancel()
+	rows, err := db.QueryContext(ctx, `SELECT NAME
```
Comment (Andrewmatilde) [inline di/di.go:104]: you may reformat the code here

File: di/di.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	db, err := sql.Open("mysql", dsn)
+	db.SetConnMaxLifetime(MYSQL_LIFE_TIME)
+	return db, err
+}
+
+// calculateDi returns total DI of specified issues
+func calculateDi(issues []Issue) float64 {
+	di := 0.0
+	for _, issue := range issues {
+		severity, ok := issue.Label["severity"]
```
Comment (Andrewmatilde) [inline di/di.go:67]: the code may write as if severity, ok := issue.Label["severity"]; !ok {....}

File: di/di_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func TestGetLabels(t *testing.T) {
+	issue := Issue{ID: 11}
+	labels, err := getLabels(db, issue)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(labels) != 3 {
+		t.Fatal()
```
Comment (Andrewmatilde) [inline di/di_test.go:40]: you may log out the error message in Fatal.

File: di/di_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+}
+
+func TestCalculateDi(t *testing.T) {
+	minorIssue := Issue{Label: map[string][]string{"severity": {"minor"}}}
+	criticalIssue := Issue{Label: map[string][]string{"severity": {"critical"}}}
+	badIssue := Issue{Label: map[string][]string{"severity": {"unknown"}}}
+
+	if calculateDI([]Issue{minorIssue, criticalIssue}) != MINOR_DI+CRITICAL_DI {
+		t.Fatal()
```
Comment (Andrewmatilde) [inline di/di_test.go:51]: ditto

File: di/di_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	if calculateDI([]Issue{badIssue, criticalIssue}) != CRITICAL_DI {
+		t.Fatal()
```
Comment (Andrewmatilde) [inline di/di_test.go:55]: ditto

File: di/di_test.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	if calculateDI([]Issue{badIssue}) != 0 {
+		t.Fatal()
```
Comment (Andrewmatilde) [inline di/di_test.go:59]: ditto

File: di/di_test.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+}
+
+func TestGetCreatedDiBetweenTime(t *testing.T) {
+	startTime := time.Date(2020, 10, 19, 0, 0, 0, 0, time.Local)
+	endTime := time.Date(2020, 10, 26, 0, 0, 0, 0, time.Local)
+	di, err := GetCreatedDiBetweenTime(db, startTime, endTime)
+	if err != nil {
+		t.Fatal()
```
Comment (Andrewmatilde) [inline di/di_test.go:69]: ditto...

File: di/di.go [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+	RepositoryID int
+	Closed       bool
+	ClosedAt     time.Time
+	CreatedAt    time.Time
+	Title        string
+	Label        map[string][]string
+}
+
+// OpenDB opens a mysql database by dsn, returns a handler
+func OpenDB(dsn string) (*sql.DB, error) {
```
Comment (zhouqiang-cl) [inline di/di.go:54]: How about move this function to an util pkg

File: di/di_test.go [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+	}
+	if len(labels) != 3 {
+		t.Fatal()
+	}
+
+}
+
+func TestCalculateDi(t *testing.T) {
+	minorIssue := Issue{Label: map[string][]string{"severity": {"minor"}}}
+	criticalIssue := Issue{Label: map[string][]string{"severity": {"critical"}}}
```
Comment (zhouqiang-cl) [inline di/di_test.go:47]: There are also have other severity, seems have 4

File: di/di.go [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+	RepositoryID int
+	Closed       bool
+	ClosedAt     time.Time
+	CreatedAt    time.Time
+	Title        string
+	Label        map[string][]string
+}
+
+// calculateDI returns total DI of specified issues
+func calculateDI(issues []Issue) float64 {
```
Comment (zhouqiang-cl) [inline di/di.go:49]: How about extract a function such as getDIofIssue()?  We may calculate much kind of DI, such as a version's DI, and Total DI

**Review by zhouqiang-cl — APPROVED** [review]
LGTM

**Pull request merged**